docbrown/db/src/csv_loader.rs

```diff
@@ -143,43 +159,44 @@ pub mod csv {
         {
             let file_path: PathBuf = path.into();
 
-            let mut csv_reader = self.csv_reader(file_path);
+            let mut csv_reader = self.csv_reader(file_path)?;
             let mut records_iter = csv_reader.deserialize::<REC>();
 
-            while let Some(rec) = records_iter.next() {
-                let record = rec.map_err(|err| CsvErr(err.into()))?;
+            for rec in records_iter {
+                let record = rec?;
                 loader(record, g)
             }
 
             Ok(())
         }
 
-        fn csv_reader(&self, file_path: PathBuf) -> csv::Reader<Box<dyn io::Read>> {
+        fn csv_reader(&self, file_path: PathBuf) -> Result<csv::Reader<Box<dyn io::Read>>, CsvErr> {
             let is_gziped = file_path
                 .file_name()
                 .and_then(|name| name.to_str())
                 .filter(|name| name.ends_with(".gz"))
                 .is_some();
 
-            let f = File::open(&file_path).expect(&format!("Can't open file {file_path:?}"));
+            let f =
+                File::open(&file_path)?;
             if is_gziped {
-                csv::ReaderBuilder::new()
+                Ok(csv::ReaderBuilder::new()
                     .has_headers(self.header)
                     .delimiter(self.delimiter)
-                    .from_reader(Box::new(BufReader::new(GzDecoder::new(f))))
+                    .from_reader(Box::new(BufReader::new(GzDecoder::new(f)))))
             } else {
-                csv::ReaderBuilder::new()
+                Ok(csv::ReaderBuilder::new()
                     .has_headers(self.header)
                     .delimiter(self.delimiter)
-                    .from_reader(Box::new(f))
+                    .from_reader(Box::new(f)))
             }
         }
 
-        pub fn load(&self) -> Result<Graph, CsvErr> {
-            let g = Graph::new(2);
-            // self.load_into(&g)?;
-            Ok(g)
-        }
+        // pub fn load(&self) -> Result<Graph, CsvErr> {
```

Comment:
we don't need this?

Reply:
This seems to be left-over code that is no longer useful? It just makes a new empty graph for some reason... I have deleted the lines rather than just commenting it out now.